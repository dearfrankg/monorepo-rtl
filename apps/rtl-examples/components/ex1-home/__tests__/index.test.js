import { render, screen } from "@testing-library/react";
import Home from "@/components/ex1-home";

/*
 * EXAMPLE1: TEST snapshot and getByRole
 * SOURCE: nextjs example: `npx create-next-app --example with-jest <app_name>`
 */
describe("ex1 <Home /> --- uses jest snapshots", () => {
  it("rendering component matches snapshot", () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });

  it("rendering component has specific header", () => {
    render(<Home />);

    const heading = screen.getByRole("heading", {
      name: /Testing Library React Examples/i,
    });

    expect(heading).toBeInTheDocument();
  });
});
