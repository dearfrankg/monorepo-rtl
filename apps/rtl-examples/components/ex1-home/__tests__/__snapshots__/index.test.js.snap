// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ex1 <Home /> --- uses jest snapshots rendering component matches snapshot 1`] = `
<div>
  <div
    style="display: flex; justify-content: center; align-items: flex-start; text-align: center;"
  >
    <main>
      <h1>
        React Testing Library Examples
      </h1>
      <span
        class="ant-typography"
      >
        <code>
          cd app_dir; yarn test
        </code>
      </span>
    </main>
  </div>
</div>
`;
